app/controllers/books_controller.rb

```diff
@@ -4,11 +4,14 @@ class BooksController < ApplicationController
 
   def index
     if params[:sort_field] == "Most popular"
-      @books = Kaminari.paginate_array(Book.most_popular).page(params[:page])
+      @books = Kaminari.paginate_array(Book.search(params[:search]).most_popular)
+                       .page(params[:page])
+
     else
       @books = Book.search(params[:search])
                    .order_by(params[:sort_field], params[:sort_order])
                    .page(params[:page])
+                   .decorate
     end
     @sortable_fields = Book.sort_options
   end
```

Comment:
You don't need to change anything but as you start using services, you'll want to do all your sorting/decorating/etc logic within the service. Just an FYI.